sw/cheri/checks/usbdev_check.cc

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define CHERIOT_NO_AMBIENT_MALLOC
+#define CHERIOT_NO_NEW_DELETE
+#define CHERIOT_PLATFORM_CUSTOM_UART
+
+#include "../../common/defs.h"
+#include "../common/usbdev-utils.hh"
+
+#include <platform-uart.hh>
+
+#define LOG(...) uart_log(uart, __VA_ARGS__)
+
+using namespace CHERI;
+
+// Shall we disconnect once we've been configured?
+//
+// - configuration indicates successful communication over the USB and the ability to send and
+//   receive data. For an FPGA, however, the host may be monitoring the USB serial connection
+//   (eg. /dev/ttyUSBx) and it may be useful to keep running and transmitting text.
+//
+// - if disconnection is not enabled then a terminator emulator may be attached to `/dev/ttyUSB<n>`
+//   and a sign-on message should be observed. Any characters entered into the terminal will
+//   be echoed on the UART output.
+static constexpr bool do_disconnect = true;
+
+static void uart_log(Capability<volatile OpenTitanUart> &uart, const char* str)
+{
+  for (; *str != '\0'; ++str)
+  {
+    uart->blocking_write(*str);
+  }
+}
+
+static void uart_logn(Capability<volatile OpenTitanUart> &uart, const char* str, size_t len)
+{
+  while (len-- > 0u)
+  {
+    uart->blocking_write(*str);
+    ++str;
+  }
+}
```

Comment:
Can't you use `write_str` instead?
https://github.com/lowRISC/sonata-system/blob/275eaab74ebec4fcf045b071eb834e586d1d28f5/sw/cheri/common/uart-utils.hh#L13